### ISSUE TITLE
Connect multiple tcam with hardwired connection?

### ISSUE THREAD

**Issue by leoshmu** [danjulio/tCam#10]
Hi, 
Is there a way to connect more than one tcam to one raspberry pi? For example would you use a serial splitting hat for the Rpi, or is there a way to hardwire it directly without relying on an additional board?

**Comment by danjulio** [comment]
Hi.  I'm assuming you mean tCam-Mini with the hardware interface.  I think that a Pi could support two cameras attached this way.  The limiting factor would be the SPI interface (https://pinout.xyz/pinout/spi).  It has (I think) 4 serial interfaces so that shouldn't be a limiting factor. 

Curious, any reason you can't just connect via Wifi?

**Comment by leoshmu** [comment]
I was going to try both hardwired and Wifi to see which is more performant. I would have guessed the hardwire approach would be better especially with 2 cameras but that may be a bad assumption. For the dual camera hardwire, with the 2nd camera be connected to the same TX/RX pins, but then the MISO/CLK etc pins would be different?

**Comment by danjulio** [comment]
You'd connect both the serial and SPI interfaces to different pins to prevent conflict on the lines driven by tCam-Mini.  Plus the interface architecture really expects only one camera per interface.

Probably 2 cameras over wifi would be fine for streaming the full 9FPS.  The upper limit is dependent on a lot of things such as signal strength and the AP performance but I've successfully streamed 6 cameras at full frame rate over my wifi here to various devices.

**Comment by leoshmu** [comment]
Sounds great, I will try with WIFI directly then.

**Comment by leoshmu** [comment]
> You'd connect both the serial and SPI interfaces to different pins to prevent conflict on the lines driven by tCam-Mini. Plus the interface architecture really expects only one camera per interface.
> 
> Probably 2 cameras over wifi would be fine for streaming the full 9FPS. The upper limit is dependent on a lot of things such as signal strength and the AP performance but I've successfully streamed 6 cameras at full frame rate over my wifi here to various devices.

Hi, something I wanted to clarify with you - can you stream more than one camera to a single device? Is there a way to do that with the python API? Or is there a limit of only one camera per computer?

**Comment by danjulio** [comment]
Yes, you can run multiple python programs, each connecting to a different camera (IP address).

You can really only connect to one camera using the desktop app, mainly because of the way it handles preferences (although on some platforms you can run multiple copies, but the last one to update preferences will always set what the preferences are next time the apps are run).
